vignettes/adtte.Rmd

```diff
@@ -0,0 +1,265 @@
+
+```{r, warning=FALSE, message=FALSE}
+library(admiral)
+library(admiralonco)
+library(dplyr)
+library(admiral.test)
+```
+
+```{r, warning=FALSE, message=FALSE, include=FALSE}
+library(lubridate)
+```
+
+# Programming Workflow
+
+* [Read in Data](#readdata)
+* [Derive Parameters (`CNSR`, `ADT`, `STARTDT`)](#parameters)
+* [Derive Analysis Value (`AVAL`)](#aval)
+* [Derive Analysis Sequence Number (`ASEQ`)](#aseq)
+* [Add ADSL Variables](#adslvars)
+
+## Read in Data {#readdata}
+
+To start, all datasets needed for the creation of the time-to-event dataset
+should be read into the environment. This will be a company specific process.
+
+For example purpose, the ADaM datasets---which are included
+in `{admiral}` and `{admiralonco}`---are used. An alternative might be to use
+`ADEVENT` as input.
+
+```{r}
+data("admiral_adsl")
+data("admiral_adrs")
+adsl <- admiral_adsl
+adrs <- admiral_adrs
+```
+
+## Derive Parameters (`CNSR`, `ADT`, `STARTDT`) {#parameters}
+
+To derive the parameter dependent variables like `CNSR`, `ADT`, `STARTDT`,
+`EVNTDESC`, `SRCDOM`, `PARAMCD`, ... the `admiral::derive_param_tte()` function can be
+used. It adds one parameter to the input dataset with one observation per
+subject. Usually it is called several times.
+
+For each subject it is determined if an event occurred. In the affirmative the
+analysis date `ADT` is set to the earliest event date. If no event occurred, the
+analysis date is set to the latest censoring date.
+
+The events and censorings are defined by the `admiral::event_source()` and the
+`admiral::censor_source()` class respectively. It defines
+
+- which observations (`filter` parameter) of a source dataset (`dataset_name`
+parameter) are potential events or censorings,
+- the value of the `CNSR` variable (`censor` parameter), and
+- which variable provides the date (`date` parameter).
+
+The date can be provided as date (`--DT` variable), datetime (`--DTM` variable),
+or character ISO-8601 date (`--DTC` variable).
+
+CDISC strongly recommends `CNSR = 0` for events and positive integers for
+censorings. `{admiral}`/`{admiralonco}` enforce this recommendation. Therefore the `censor`
+parameter is available for `admiral::censor_source()` only. It is defaulted to `1`.
+
+The `dataset_name` parameter expects a character value which is used as an
+identifier. The actual data which is used for the derivation of the parameter is
+provided via the `source_datasets` parameter of `admiral::derive_param_tte()`. It expects
+a named list of datasets. The names correspond to the identifiers specified for
+the `dataset_name` parameter. This allows to define events and censoring
+independent of the data.
+
+### Pre-Defined Time-to-Event Source Objects
+
+The table below shows all pre-defined `tte_source` objects which should cover the most
+common oncology use cases.
+
+```{r echo=FALSE}
+knitr::kable(admiral:::list_tte_source_objects(package = "admiralonco"))
```

Comment:
No need for `:::` as `list_tte_sources()` is now an exported function of {admiral}.